ast/ast.hpp

```diff
@@ -27,8 +27,11 @@ struct WhileLoop;
 struct ForLoop;
 struct IfStatement;
 struct Expression;
-struct UnaryExpression;
-struct BinaryExpression;
+// struct UnaryExpression;
+// struct BinaryExpression;
+struct Primitive;
```

Comment:
It is not clear to me what this `Primitive` represents exactly. It appears to only hold a single token.

Reply:
Yes. It is a wrapper for token to be used as an expression

Reply:
Ok, that's fine for the scope of this PR, and after that we can see how relevant it is with the design I propose (which we will discuss in next meeting)